app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -893,6 +894,8 @@ protected void onNewIntent(Intent intent) {
                 // photo was deleted and camera is launched again to capture a new photo
                 dispatchTakePictureIntent();
             }
+        } else if (action.equals(ACTION_MESSAGE_PUSH_NOTIFICATION)) {
```

Comment:
cool, structurally I think it might be better to handle this case in the preceding if-else as that is for discerning actions. Works well though 👍